File: tests-python/web_selenium/tests/ajax_test.py
import allure
from web_selenium.conftest import driver
from web_selenium import pages


@allure.link("https://www.selenium.dev/selenium/web/web-form.html", name="Target webpage")
@allure.issue("JIRA-123")
@allure.testcase("TEST-456")
@allure.parent_suite("Web interface (Selenium)")
@allure.suite("Ajax")
@allure.epic("Web interface (Selenium)")
#@allure.story("Ajax")
@allure.feature("Ajax")
def test_ajax_response(driver):
    """
    Testing a webpage using AJAX.

    Test using WebDriverWait.until()
    """
    driver.get("http://harmin-demo.gitlab.io/reports/web/ajax.html")
    allure.attach(
        driver.get_screenshot_as_png(),
        name="Initial page",
        attachment_type=allure.attachment_type.PNG
    )
    ajax = pages.AjaxPage(driver)
    ajax.click()
    allure.attach(
        driver.get_screenshot_as_png(),
        name="Trigger event",
        attachment_type=allure.attachment_type.PNG
    )
    ajax.verify()
    allure.attach(
        driver.get_screenshot_as_png(),
        name="Verify event result",
        attachment_type=allure.attachment_type.PNG
    )

File: tests-java/src/test/java/web_selenium/AjaxTest.java
package web_selenium;

import java.io.ByteArrayInputStream;
import java.net.MalformedURLException;
import java.net.URL;

import org.apache.commons.lang3.StringUtils;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import io.qameta.allure.Allure;
import io.qameta.allure.Description;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.edge.EdgeOptions;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxOptions;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.OutputType;
import org.openqa.selenium.TakesScreenshot;

import web_selenium.AjaxPage;

public class AjaxTest {

    private WebDriver driver;

    @BeforeEach
    public void setup() {
        this.driver = Common.getDriver();
    }

    /**
     * Testing a webpage using AJAX.
     *
     * Test using <span style="font-family:'Courier New'">WebDriverWait.until()</span>
     */
    @Test
    @Description
    public void ajax_response() {
        Allure.getLifecycle().updateTestCase(tr -> tr.getLabels().removeIf(label -> "suite".equals(label.getName())));
        Allure.epic("Web interface (Selenium)");
        // Allure.story("Web Form");
        Allure.suite("Web interface (Selenium)");
        Allure.feature("Ajax page");
        this.driver.get("http://harmin-demo.gitlab.io/reports/web/ajax.html");
        byte[] buffer = ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
        Allure.addAttachment("Initial page", new ByteArrayInputStream(buffer));
        AjaxPage page = new AjaxPage(this.driver);
        page.click();
        buffer = ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
        Allure.addAttachment("Trigger event", new ByteArrayInputStream(buffer));
        page.wait_ajax();
        buffer = ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
        Allure.addAttachment("Verify event result", new ByteArrayInputStream(buffer));
        page.verify();
    }

    @AfterEach
    public void teardown() {
        try {
            this.driver.quit();
        } catch (Exception e) { }
    }
}
